packages/components/upload/src/use-handlers.ts

```diff
@@ -0,0 +1,184 @@
+import { ref, watch } from 'vue'
+import { debugWarn, throwError } from '@element-plus/utils'
+import { useDeprecated } from '@element-plus/hooks'
+import { genFileId } from './upload'
+import type { ShallowRef } from 'vue'
+import type {
+  UploadContentProps,
+  UploadContentInstance,
+} from './upload-content'
+import type {
+  UploadRawFile,
+  UploadFile,
+  UploadProps,
+  UploadStatus,
+  UploadFiles,
+} from './upload'
+
+const SCOPE = 'ElUpload'
+
+const revokeObjectURL = (file: UploadFile) => {
+  if (file.url?.startsWith('blob:')) {
+    URL.revokeObjectURL(file.url)
+  }
+}
+
+export const useHandlers = (
+  props: UploadProps,
+  uploadRef: ShallowRef<UploadContentInstance | undefined>
+) => {
+  const uploadFiles = ref<UploadFiles>([])
+
+  const getFile = (rawFile: UploadRawFile) =>
+    uploadFiles.value.find((file) => file.uid === rawFile.uid)
+
+  function abort(file: UploadFile) {
+    uploadRef.value?.abort(file)
+  }
+
+  function clearFiles(
+    /** @default ['ready', 'uploading', 'success', 'fail'] */
+    status: UploadStatus[] = ['ready', 'uploading', 'success', 'fail']
```

Comment:
**Suggestion:** rename the parameter `status` to `states` to avoid name collision.